cmd/ascii2der/scanner.go

```diff
@@ -284,6 +284,51 @@ again:
 		if s.pos.Offset+1 < len(s.text) && s.text[s.pos.Offset+1] == '"' {
 			return s.parseUTF32String()
 		}
+	case 'b':
+		if s.pos.Offset+1 < len(s.text) && s.text[s.pos.Offset+1] == '`' {
+			s.advance() // Skip the b.
+			s.advance() // Skip the `.
+			bitStr, ok := s.consumeUpTo('`')
+			if !ok {
+				return token{}, &parseError{s.pos, errors.New("unmatched `")}
+			}
+
+			// Leading byte is the number of "extra" bits at the end.
+			var bitCount int
+			var sawPipe bool
+			bits := []byte{0}
+			for i, r := range bitStr {
+				switch r {
+				case '0', '1':
+					if bitCount%8 == 0 {
+						bits = append(bits, 0)
+					}
+					if r == '1' {
+						bits[len(bits)-1] |= 1 << uint(bitCount%8)
+					}
+					bitCount++
+				case '|':
+					if sawPipe {
+						return token{}, &parseError{s.pos, errors.New("duplicate '")}
+					}
+
+					// Ensure that there aren't actually more than |padding| bits past the |.
+					padding := (len(bits)-1)*8 - bitCount
+					if len(bitStr)-i > padding+1 {
+						return token{}, &parseError{s.pos, errors.New("more than 7 bits after '")}
```

Comment:
The comment says `padding`, the code says `padding+1`, while the error says 7. The error string seems wrong. `padding+1` I guess is because this is really checking `len(bitStr)-i-1` to account for the pipe? That might be a bit easier to read.

Reply:
Added some more variables to clear things up. I wasn't deeply happy with how I organized this code.